server/tests/replicated_loglet.rs

```diff
@@ -175,4 +182,168 @@ mod tests {
+                    appenders.spawn({
+                        let bifrost = test_env.bifrost.clone();
+                        let cancel_appenders = cancel_appenders.clone();
+                        let tc = tc.clone();
+                        async move {
+                            tc.run_in_scope("append", None, async move {
+                                let mut i = 1;
+                                let mut committed = Vec::new();
+                                while !cancel_appenders.is_cancelled() {
+                                    let offset = bifrost
+                                        .append(
+                                            log_id,
+                                            format!("appender-{}-record{}", appender_id, i),
+                                        )
+                                        .await?;
+                                    i += 1;
+                                    committed.push(offset);
+                                }
+                                Ok(committed)
+                            })
+                            .await
+                        }
+                    });
+                }
+
+                let mut sealer_handle: JoinHandle<googletest::Result<()>> = tokio::task::spawn({
+                    let tc = tc.clone();
+                    let (bifrost, metadata_writer, metadata_store_client) = (
+                        test_env.bifrost.clone(),
+                        test_env.metadata_writer.clone(),
+                        test_env.metadata_store_client.clone()
+                    );
+
+                    async move {
+                        let cancellation_token = cancellation_token();
+
+                        let mut chain = metadata.updateable_logs_metadata().map(|logs| logs.chain(&log_id).expect("a chain to exist"));
+
+                        let bifrost_admin = restate_bifrost::BifrostAdmin::new(
+                            &bifrost,
+                            &metadata_writer,
+                            &metadata_store_client,
+                        );
+
+                        tc.run_in_scope("sealer", None, async move {
+                            let mut last_loglet_id = None;
+
+                            while !cancellation_token.is_cancelled() {
+                                tokio::time::sleep(std::time::Duration::from_secs(10)).await;
+
+                                let mut params = ReplicatedLogletParams::deserialize_from(
+                                    chain.live_load().tail().config.params.as_ref(),
+                                )?;
+                                if last_loglet_id == Some(params.loglet_id) {
+                                    eprintln!("Skipping seal as metadata has not caught up from the last seal (version={})", metadata.logs_version());
+                                    continue;
+                                }
```

Comment:
Interesting! do you actually see this happening? IIRC, seal_and_extend_chain will write the updated metadata to our metadata manager before it returns, so we shouldn't really see this happening.

Reply:
no, i think this only happened when operator error meant that the seal didnt finish (eg when i didnt increment the segment index). what ill do instead is fail here